File: src/main/java/net/tassia/pancake/util/Base64.java
package net.tassia.pancake.util;

/**
 * Utility class for Base64 encoding/decoding.
 *
 * @since Pancake 1.0
 * @author Tassilo
 */
public final class Base64 {

	/**
	 * Encodes the data using Base64.
	 *
	 * @param data the data to encode
	 * @return a base64 string
	 */
	public static String encode(byte[] data) {
		return java.util.Base64.getEncoder().encodeToString(data);
	}

	/**
	 * Decodes the Base64 string.
	 *
	 * @param base64 the base64 string
	 * @return the decoded data
	 */
	public static byte[] decode(String base64) {
		return java.util.Base64.getDecoder().decode(base64);
	}



	/**
	 * Static class.
	 */
	private Base64() {
	}

}

File: src/main/java/net/tassia/pancake/util/Base64.kt
package net.tassia.pancake.util

import java.nio.charset.StandardCharsets
import java.util.Base64

/**
 * Allows to encode/decode byte arrays into Base64.
 *
 * @since Pancake 1.0
 * @author Tassilo
 */
object Base64 {

	/**
	 * Encodes the given byte array into a Base64 string.
	 *
	 * @param data the bytes to encode
	 * @return the Base64 string
	 */
	fun encode(data: ByteArray): String {
		return Base64.getEncoder().encode(data).toString(StandardCharsets.US_ASCII)
	}

	/**
	 * Decodes the given Base64 string into its source byte array.
	 *
	 * @param base64 the Base64 string
	 * @return the source bytes
	 */
	fun decode(base64: String): ByteArray {
		return Base64.getDecoder().decode(base64)
	}

}
